src/javaharness/java/arcs/android/demo/service/ArcsAutofillService.java

```diff
@@ -56,24 +62,17 @@ public void onFillRequest(
 
     RemotePec remotePec = remotePecProvider.get();
 
-    AutofillParticle autofillParticle =
-        new AutofillParticle(
-            jsonParser,
-            node.get(),
-            (autofillId, suggestion) -> {
-              Dataset.Builder dataset = new Dataset.Builder();
-              dataset.setValue(
-                  autofillId, AutofillValue.forText(suggestion), createRemoteView(suggestion));
-
-              FillResponse fillResponse =
-                  new FillResponse.Builder().addDataset(dataset.build()).build();
-
-              callback.onSuccess(fillResponse);
-              remotePec.shutdown();
-            });
-
+    AutofillParticle autofillParticle = new AutofillParticle(node.get());
     // Start up an Arcs remote PEC and arc with the autofill particle.
     remotePec.runArc("AndroidAutofill", autofillParticle);
+    autofillRenderer.addCallback(
+        // TODO: Use provided slot id instead.
+        remotePec.getArcId().toString(), // autofillParticle.getId(),
+        getPackageName(),
+        node.get().getAutofillId(),
+        callback,
+        () -> { remotePec.shutdown(); });
```

Comment:
Rather than having two callback functions, you could just use the one:

```
fillResponse -> {
  callback.onSuccess(fillResponse);
  remotePec.shutdown();
}
```

Reply:
done in #3611